File: spark-usage/src/test/java/info/szadkowski/bqissue/fixes/DummyCredentials.java
package info.szadkowski.bqissue.fixes;

import com.google.cloud.spark.bigquery.repackaged.com.google.auth.Credentials;

import java.io.IOException;
import java.net.URI;
import java.util.List;
import java.util.Map;

import static java.util.Collections.emptyMap;

public class DummyCredentials extends Credentials {
    @Override
    public String getAuthenticationType() {
        throw new UnsupportedOperationException();
    }

    @Override
    public Map<String, List<String>> getRequestMetadata(URI uri) throws IOException {
        return emptyMap();
    }

    @Override
    public boolean hasRequestMetadata() {
        throw new UnsupportedOperationException();
    }

    @Override
    public boolean hasRequestMetadataOnly() {
        throw new UnsupportedOperationException();
    }

    @Override
    public void refresh() {
        throw new UnsupportedOperationException();
    }
}

File: src/test/kotlin/info/szadkowski/bqissue/DummyCredentials.kt
package info.szadkowski.bqissue

import com.google.auth.Credentials
import java.net.URI

class DummyCredentials : Credentials() {
    override fun getAuthenticationType(): String {
        TODO("Not yet implemented")
    }

    override fun getRequestMetadata(uri: URI?): MutableMap<String, MutableList<String>> {
        return mutableMapOf()
    }

    override fun hasRequestMetadata(): Boolean {
        TODO("Not yet implemented")
    }

    override fun hasRequestMetadataOnly(): Boolean {
        TODO("Not yet implemented")
    }

    override fun refresh() {
        TODO("Not yet implemented")
    }
}
